### PR TITLE
docs: add troubleshooting guide

### PR DESCRIPTION
**Pull request by alexeagle**
Ported from rules_nodejs wiki and removed rootDirs stuff

### CODE REVIEW

File: docs/rules.md [hunk 1]
```diff
@@ -160,29 +160,19 @@ package, typically loaded from `@npm//typescript:package_json.bzl`.
 Unlike bare `tsc`, this rule understands the Bazel interop mechanism (Providers)
 so that this rule works with others that produce or consume TypeScript typings (`.d.ts` files).
 
-One of the benefits of using ts_project is that it understands Bazel Worker Protocol which makes
-JIT overhead one time cost. Worker mode is on by default to speed up build and typechecking process.
+One of the benefits of using ts_project is that it understands the [Bazel Worker Protocol]
```
Comment (jbedard) [inline docs/rules.md:163]: Was "Bazel Worker Protocol" meant to be a link?

File: docs/rules.md [hunk 2]
```diff
@@ -160,29 +160,19 @@ package, typically loaded from `@npm//typescript:package_json.bzl`.
...
-- Your tsconfig settings for `outDir` and `declarationDir` are ignored.
-  Bazel requires that the `outDir` (and `declarationDir`) be set beneath
-  `bazel-out/[target architecture]/bin/path/to/package`.
-- Bazel expects that each output is produced by a single rule.
-  Thus if you have two `ts_project` rules with overlapping sources (the same `.ts` file
-  appears in more than one) then you get an error about conflicting `.js` output
-  files if you try to build both together.
-  Worse, if you build them separately then the output directory will contain whichever
-  one you happened to build most recently. This is highly discouraged.
+If you have problems getting your `ts_project` to work correctly, read the dedicated
```
Comment (jbedard) [inline docs/rules.md:172]:
```suggestion
If you have problems getting your `ts_project` working correctly, read the dedicated
```

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    Matched by include pattern '**/*' in 'tsconfig.json'
+```
+
+The `--explainFiles` flag in TS 4.2 also gives information about why a given file was added to the program.
+
+## Module not resolved
+
+Use the `--traceResolution` flag to `tsc` to understand where TypeScript looked for the file.
+
+Verify is that there is actually a `.d.ts` file for TypeScript to resolve. Check that the dependency library has the `declarations = True` flag set, and that the `.d.ts` files appear where you expect them under `bazel-out`.
```
Comment (jbedard) [inline docs/troubleshooting.md:58]:
```suggestion
Verify that there is actually a `.d.ts` file for TypeScript to resolve. Check that the dependency library has the `declarations = True` flag set, and that the `.d.ts` files appear where you expect them under `bazel-out`.
```

File: ts/defs.bzl [hunk 4]
```diff
@@ -83,29 +83,19 @@ def ts_project(
     Unlike bare `tsc`, this rule understands the Bazel interop mechanism (Providers)
     so that this rule works with others that produce or consume TypeScript typings (`.d.ts` files).
 
-    One of the benefits of using ts_project is that it understands Bazel Worker Protocol which makes
-    JIT overhead one time cost. Worker mode is on by default to speed up build and typechecking process.
+    One of the benefits of using ts_project is that it understands the [Bazel Worker Protocol]
```
Comment (jbedard) [inline ts/defs.bzl:86]: Was "Bazel Worker Protocol" meant to be a link?

**Review by jbedard — APPROVED** [review]

File: ts/defs.bzl (hunk 4 above)
Comment (alexeagle) [reply ts/defs.bzl:86]: it is, using the "reference-style links" markdown syntax  https://www.markdownguide.org/basic-syntax/#reference-style-links

Comment (jbedard) [reply ts/defs.bzl:86]: TIL... 👍

**Pull request merged**